File: backend/app/schemas/m365_intelligence.py
"""Schemas — inteligencia M365 (correo y repositorios)."""

from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class M365SuggestedAction(BaseModel):
    id: str
    label: str
    action_type: str
    href: str | None = None


class M365MailIntelligenceResponse(BaseModel):
    message_id: str
    classification: str
    classification_label: str
    classification_confidence: int
    matched_signals: list[str] = Field(default_factory=list)
    summary: str = ""
    sentiment: str = "neutral"
    priority_score: int = 50
    priority_label: str = "Normal"
    vendor: str | None = None
    client: str | None = None
    amount: float | None = None
    currency: str | None = None
    dgcp_process_code: str | None = None
    products: list[str] = Field(default_factory=list)
    document_type: str | None = None
    dates: list[str] = Field(default_factory=list)
    risks: list[str] = Field(default_factory=list)
    opportunities: list[str] = Field(default_factory=list)
    suggested_actions: list[M365SuggestedAction] = Field(default_factory=list)


class M365CalendarEventCreate(BaseModel):
    subject: str
    start: str
    end: str
    location: str = ""
    body: str = ""
    attendees: list[str] = Field(default_factory=list)
    is_online: bool = False


class M365CalendarEventUpdate(BaseModel):
    subject: str | None = None
    start: str | None = None
    end: str | None = None
    location: str | None = None
    body: str | None = None


class M365CalendarActionResponse(BaseModel):
    ok: bool
    message: str
    event_id: str | None = None


class M365RepositoryFileItem(BaseModel):
    id: uuid.UUID
    name: str
    source: str
    parent_path: str = ""
    document_category: str
    document_category_label: str
    document_type: str
    classification_confidence: int
    tags: list[str] = Field(default_factory=list)
    company_key: str | None = None
    is_folder: bool = False
    mime_type: str | None = None
    size_bytes: int | None = None
    web_url: str | None = None
    download_url: str | None = None
    graph_item_id: str
    synced_at: datetime | None = None


class M365RepositoryListResponse(BaseModel):
    items: list[M365RepositoryFileItem]
    total: int
    categories: dict[str, int] = Field(default_factory=dict)
    connected: bool = True
    message: str = ""


class M365RepositorySyncResponse(BaseModel):
    ok: bool
    synced: int
    classified: int
    message: str


class M365TemplateField(BaseModel):
    key: str
    label: str
    value: str
    source: str = ""
    confidence: float = 0.0


class M365TemplatePreviewResponse(BaseModel):
    template_type: str
    template_label: str
    fields: list[M365TemplateField] = Field(default_factory=list)
    missing_fields: list[str] = Field(default_factory=list)
    generate_enabled: bool = False


class M365TemplateGenerateResponse(BaseModel):
    ok: bool
    filename: str
    content_base64: str
    format: str = "docx"
    message: str = ""


class M365SemanticSearchResponse(BaseModel):
    query: str
    hits: list[dict] = Field(default_factory=list)
    total: int = 0
    qdrant_ready: bool = False
    message: str = ""


class M365UploadResponse(BaseModel):
    ok: bool
    message: str
    item_id: str | None = None
    web_url: str | None = None

File: frontend/src/lib/m365-intelligence.ts
export interface M365SuggestedAction {
  id: string;
  label: string;
  action_type: string;
  href?: string | null;
}

export interface M365MailIntelligence {
  message_id: string;
  classification: string;
  classification_label: string;
  classification_confidence: number;
  matched_signals: string[];
  summary: string;
  sentiment: string;
  priority_score: number;
  priority_label: string;
  vendor?: string | null;
  client?: string | null;
  amount?: number | null;
  currency?: string | null;
  dgcp_process_code?: string | null;
  products: string[];
  document_type?: string | null;
  dates: string[];
  risks: string[];
  opportunities: string[];
  suggested_actions: M365SuggestedAction[];
}

export interface M365RepositoryFile {
  id: string;
  name: string;
  source: string;
  parent_path: string;
  document_category: string;
  document_category_label: string;
  document_type: string;
  classification_confidence: number;
  tags: string[];
  company_key?: string | null;
  is_folder: boolean;
  mime_type?: string | null;
  size_bytes?: number | null;
  web_url?: string | null;
  download_url?: string | null;
  graph_item_id: string;
}

export interface M365TemplatePreview {
  template_type: string;
  template_label: string;
  fields: Array<{ key: string; label: string; value: string; source: string; confidence: number }>;
  missing_fields: string[];
  generate_enabled: boolean;
}

export const REPOSITORY_CATEGORIES: { id: string; label: string; color: string }[] = [
  { id: "legal", label: "Legales", color: "bg-red-500/15 text-red-700" },
  { id: "formulario", label: "Formularios", color: "bg-blue-500/15 text-blue-700" },
  { id: "constitucion", label: "Constitución", color: "bg-purple-500/15 text-purple-700" },
  { id: "plantilla", label: "Plantillas", color: "bg-indigo-500/15 text-indigo-700" },
  { id: "contrato", label: "Contratos", color: "bg-orange-500/15 text-orange-700" },
  { id: "licitacion", label: "Licitaciones", color: "bg-emerald-500/15 text-emerald-700" },
  { id: "proveedor", label: "Proveedores", color: "bg-cyan-500/15 text-cyan-700" },
  { id: "financiero", label: "Finanzas", color: "bg-amber-500/15 text-amber-700" },
  { id: "ficha_tecnica", label: "Fichas técnicas", color: "bg-slate-500/15 text-slate-700" },
  { id: "cotizacion", label: "Cotizaciones", color: "bg-teal-500/15 text-teal-700" },
  { id: "general", label: "General", color: "bg-muted text-muted-foreground" },
];
